### PR TITLE
Fixes #32389 - Quote taxonomies in search query

### PR DESCRIPTION
**Pull request by adiabramovitch**
Fixes bug introduced by 377ff66, where we forgot to quote search query strings in links on host details page.

### CODE REVIEW

File: app/helpers/host_description_helper.rb [hunk 1]
```diff
@@ -76,7 +76,7 @@ def base_host_overview_fields(host)
     fields << { :field => [_("Host group"), link_to(host.hostgroup, hosts_path(:search => %{hostgroup_title = "#{host.hostgroup}"}))], :priority => 1000 } if host.hostgroup.present?
     fields << { :field => [_("Boot time"), (boot_time = host&.reported_data&.boot_time) ? date_time_relative(boot_time) : _('Not reported')], :priority => 1100 }
     fields << { :field => [_("Location"), link_to(host.location.title, hosts_path(:search => "location = #{host.location}"))], :priority => 1200 } if host.location.present?
```
Comment (ekohl) [inline app/helpers/host_description_helper.rb]: Would you mind fixing this line too? Locations can have the exact same problem

**Review by ekohl — DISMISSED** [review]
In the commit message I would describe the solution, not the problem. So rather "Quote taxonomies in search query" than "Incorrect search query".

Also, it was 377ff667bbe3c2df8cd0831d9fe60bb9b316d680 that introduced this regression so affects 1.24 - 2.4. Needs a cherry pick.

File: app/helpers/host_description_helper.rb (hunk 1 above)
Comment (adiabramovitch) [reply app/helpers/host_description_helper.rb]: Sure, thank you for noticing. 
can you explain what do you mean by cherry-picking? (:

Comment (ekohl) [reply app/helpers/host_description_helper.rb]: Cherry picking is the process we use to get fixes into older releases.

What I did was `git blame -L 79,79 app/helpers/host_description_helper.rb` which pointed me to 079723bee1. Then I used `git blame 079723bee1^ app/helpers/host_description_helper.rb` (`^` means the parent of that commit). That showed 377ff667bb as the commit that changed it last. There I could see it wasn't a problem in the old version. Then you can use `git tag --contains 377ff667bb` which pointed me to 1.24.0 as the oldest release.

With that, I knew that this affects older releases and this bugfix would be beneficial for users. It's mostly up to the core maintainers to do the actual cherry pick, but understanding the process behind it never hurts.

**Review by ezr-ondrej — APPROVED** [review]
Works for me :+1:

**Comment by ezr-ondrej** [comment]
Thanks @adiabramovitch and @ekohl for the review! :+1:

**Comment by ekohl** [comment]
Indeed, thanks @adiabramovitch!

@ezr-ondrej I see this wasn't cherry picked yet. Will you take care of this?

**Comment by ezr-ondrej** [comment]
I'm on it.

**Comment by ezr-ondrej** [comment]
CP 2.5: #8501
CP 2.4: #8502

@adiabramovitch just FYI, what I did to cherry-pick (apply only this change to older version of code, without bringing another changes) is:
`git checkout 2.5-stable`
`git checkout -b new_branch_name`
`git cherry-pick -x <commit-hash>` commit hash can be found here in <reviewer> merged commit <commit-hash> into develop.
`git push -u origin new_branch_name`
Create a PR from `new_branch_name`, just don't forget to change target branch from develop to proper stable branch, in this case `2.5-stable`

**Comment by ekohl** [comment]
> `git checkout 2.5-stable`
> `git checkout -b new_branch_name`

You can do this in a single command.
`git checkout -b new_branch_name 2.5-stable`

>  `git push -u origin new_branch_name`

You can use `HEAD` for the current branch: `git push -u origin HEAD`. In fact, I have an alias in git  for this:
```
git config --global alias.ph "!git push $(whoami) HEAD --set-upstream --quiet"
```

Note that I always use my username as the remote instead of origin, which is why I have `$(whoami)`

>  Create a PR from `new_branch_name`, just don't forget to change target branch from develop to proper stable branch, in this case `2.5-stable`

I use [hub](https://github.com/github/hub) for this:
`hub pull-request -b 2.5-stable`

Here my alias is:
```
git config --global alias.pr "!hub pull-request"
```

Then I also combine these into a single alias:
```
git config --global alias.phr "!git ph && git pr"
```

With that, I can use `git phr` to push the current branch and create a PR from the terminal. I can use options like `git phr -b target-branch`

So a script can be:
```bash
#!/bin/bash
BRANCH=$1
COMMIT=$2
REMOTE=${3:-origin}

if [[ -z $BRANCH ]] || [[ -z $COMMIT ]] ; then
  echo "Usage: $0 BRANCH COMMIT [REMOTE]"
  exit 1
fi

git checkout -b ${BRANCH}-pick-${COMMIT}
git cherry-pick -x $COMMIT
git push -u $REMOTE HEAD
hub pull-request -b $BRANCH --no-edit
```

In the above example I didn't use the aliases to be explicit, but obviously it can be shortened if you do.

**Pull request merged**